Camera/CamerasPhantom/camera_phantom_interface.cc

```diff
@@ -0,0 +1,262 @@
+        available_grabbers_[ix]->setString<Euresys::StreamModule>("StatisticsSamplingSelector", "LastSecond");
+        available_grabbers_[ix]->setString<Euresys::StreamModule>("LUTConfiguration", "M_10x8");
+    }
+
+    for (size_t i = 0; i < nb_grabbers_; ++i)
+        available_grabbers_[i]->setInteger<Euresys::StreamModule>("StripeOffset",
+                                                                  params.at<std::vector<unsigned int>>("Offset")[i]);
+
+    if (params.has("TriggerMode"))
+        available_grabbers_[0]->setString<Euresys::RemoteModule>("TriggerMode", params.at<std::string>("TriggerMode"));
+    available_grabbers_[0]->setString<Euresys::RemoteModule>("TriggerSource", params.at<std::string>("TriggerSource"));
+
+    std::string control_mode = params.at<std::string>("TriggerSource") == "SWTRIGGER" ? "RC" : "EXTERNAL";
+    available_grabbers_[0]->setString<Euresys::DeviceModule>("CameraControlMethod", control_mode);
+
+    if (params.has("TriggerSelector"))
+        available_grabbers_[0]->setString<Euresys::RemoteModule>("TriggerSelector",
+                                                                 params.at<std::string>("TriggerSelector"));
+
+    if (params.at<std::string>("TriggerSource") == "SWTRIGGER")
+    {
+        available_grabbers_[0]->setInteger<Euresys::DeviceModule>("CycleMinimumPeriod",
+                                                                  params.at<unsigned int>("CycleMinimumPeriod"));
+        available_grabbers_[0]->setString<Euresys::DeviceModule>("ExposureReadoutOverlap", "True");
+        available_grabbers_[0]->setString<Euresys::DeviceModule>("ErrorSelector", "All");
+    }
+    available_grabbers_[0]->setFloat<Euresys::RemoteModule>("ExposureTime", params.at<float>("ExposureTime"));
+    available_grabbers_[0]->setString<Euresys::RemoteModule>("BalanceWhiteMarker",
+                                                             params.at<std::string>("BalanceWhiteMarker"));
+
+    available_grabbers_[0]->setFloat<Euresys::RemoteModule>("Gain", params.at<float>("Gain"));
+    available_grabbers_[0]->setString<Euresys::RemoteModule>("GainSelector", params.at<std::string>("GainSelector"));
+}
+
+void EHoloGrabberInt::init(unsigned int nb_buffers)
+{
+    nb_buffers_ = nb_buffers;
+    size_t frame_size = width_ * height_ * depth_;
+
+    // Allocate buffers in pinned memory
+    // Learn more about pinned memory:
+    // https://developer.nvidia.com/blog/how-optimize-data-transfers-cuda-cc/.
+
+    uint8_t* device_ptr;
+
+    cudaError_t alloc_res = cudaHostAlloc(&ptr_, frame_size * buffer_part_count_ * nb_buffers_, cudaHostAllocMapped);
+    cudaError_t device_ptr_res = cudaHostGetDevicePointer(&device_ptr, ptr_, 0);
+    if (alloc_res != cudaSuccess || device_ptr_res != cudaSuccess)
+        Logger::camera()->error("Could not allocate buffers.");
+
+    float prog = 0.0;
+    for (size_t i = 0; i < nb_buffers; ++i)
```

Comment:
There is a similar funciton `progress_bar` in the `cli.cc` file, maybe merge them ?